Rock.Encryption/Async/SynchronousAsyncCrypto.cs

```diff
@@ -0,0 +1,213 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace RockLib.Encryption.Async
+{
+    /// <summary>
+    /// An implementation of <see cref="IAsyncCrypto"/> that delegates all functionality
+    /// to an instance of <see cref="ICrypto"/>. All tasks returned by its methods are
+    /// guaranteed to be either completed or faulted.
+    /// </summary>
+    public sealed class SynchronousAsyncCrypto : IAsyncCrypto
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SynchronousAsyncCrypto"/> class.
+        /// </summary>
+        /// <param name="crypto">
+        /// The instance of <see cref="ICrypto"/> that performs the actual work
+        /// done by the methods of this instance of <see cref="SynchronousAsyncCrypto"/>.
+        /// </param>
+        public SynchronousAsyncCrypto(ICrypto crypto)
+        {
+            Crypto = crypto;
+        }
+
+        /// <summary>
+        /// Gets the instance of <see cref="ICrypto"/> that performs the actual work
+        /// done by the methods of this instance of <see cref="SynchronousAsyncCrypto"/>.
+        /// </summary>
+        public ICrypto Crypto { get; }
+
+        /// <summary>
+        /// Synchronously encrypts the specified plain text.
```

Comment:
Async not Sync

Reply:
The comment is accurate. This implementation does it synchronously, returning an already-completed task.